src/runtime/TemporalObject.h

```diff
@@ -549,11 +556,42 @@ class TemporalPlainMonthDayObject : public Temporal {
 class TemporalTimeZoneObject : public Temporal {
 public:
     explicit TemporalTimeZoneObject(ExecutionState& state);
-    explicit TemporalTimeZoneObject(ExecutionState& state, Object* proto);
+    explicit TemporalTimeZoneObject(ExecutionState& state, Object* proto, std::string identifier = "", const Value& offsetNanoseconds = Value());
 
     static std::map<TemporalObject::DateTimeUnits, int> getISOPartsFromEpoch(ExecutionState& state, const Value& epochNanoseconds);
-    static Value getOffsetNanosecondsFor(ExecutionState& state, const Value& timeZone, const Value& instant);
+    static Value createTemporalTimeZone(ExecutionState& state, const std::string& identifier, Optional<Object*> newTarget = nullptr);
+    static bool isValidTimeZoneName(const std::string& timeZone);
+    static std::string canonicalizeTimeZoneName(const std::string& timeZone);
+    static std::string formatTimeZoneOffsetString(long long offsetNanoseconds);
+    static std::string defaultTimeZone();
+    static Value toTemporalTimeZone(ExecutionState& state, const Value& temporalTimeZoneLike);
+    static std::map<TemporalObject::DateTimeUnits, int> getIANATimeZoneDateTimeParts(ExecutionState& state, const Value& epochNanoseconds);
+    static int64_t getOffsetNanosecondsFor(ExecutionState& state, const Value& timeZone, const Value& instant);
+    static Value builtinTimeZoneGetOffsetStringFor(ExecutionState& state, const Value& timeZone, const Value& instant);
     static Value builtinTimeZoneGetPlainDateTimeFor(ExecutionState& state, const Value& timeZone, const Value& instant, const Value& calendar);
+    static Value getIANATimeZoneOffsetNanoseconds(ExecutionState& state, const Value& epochNanoseconds, const std::string& timeZoneIdentifier);
+    static ValueVector getIANATimeZoneEpochValue(ExecutionState& state, const std::string& timeZoneIdentifier, int year, int month, int day, int hour, int minute, int second, int millisecond, int microsecond, int nanosecond);
+    static Value getIANATimeZoneNextTransition(ExecutionState& state, const Value& epochNanoseconds, std::string timeZoneIdentifier);
+    static Value getIANATimeZonePreviousTransition(ExecutionState& state, const Value& epochNanoseconds, std::string timeZoneIdentifier);
+
+    bool isTemporalTimeZoneObject() const override
+    {
+        return true;
+    }
+
+    const std::string& getIdentifier() const
+    {
+        return m_identifier;
+    }
+    const Value& getOffsetNanoseconds() const
+    {
+        return m_offsetNanoseconds;
+    }
+
+private:
+    static Value getIANATimeZoneTransition(ExecutionState& state, BigInt* rightNanos, BigInt* rightOffsetNs, BigInt* leftNanos, BigInt* leftOffsetNs, BigInt* cap, const std::string& timeZoneIdentifier);
+    std::string m_identifier;
```

Comment:
`TemporalTimeZoneObject` has a `std::string` member now.
But one thing to note is that the destructor of `TemporalTimeZoneObject` is never called when its instance is destroyed.
GC just reclaims the size of heap object, so the destructure of `m_identifier` would not be called either which may lead to another memory leak. Pleack check it